db/migrate/20181201000400_create_supplier_categories.rb

```diff
@@ -0,0 +1,28 @@
+class CreateSupplierCategories < ActiveRecord::Migration
+  class FinancialTransactionClass < ActiveRecord::Base
+  end
+  class SupplierCategory < ActiveRecord::Base
+  end
+  class Supplier < ActiveRecord::Base
+  end
+
+  def change
+    create_table :supplier_categories do |t|
+      t.string :name, null: false
+      t.string :description
+      t.references :financial_transaction_class, null: false
```

Comment:
Just to be sure: can there be zero transaction classes?
I can image supplier categories to make sense without them.

Reply:
i think that the class is mandatory - otherwise the FinanceReport (you can find it in the [Austrian fork](https://github.com/foodcoopsat/foodsoft/) for now) can not assign the invoices to the correct sum
example: you have three supplier_categories ("food", "rent", "utils") and two finacial_transaction_classes ("food", "membership"). "rent" is used for the rent and electricity of the room, "utils" is used when new fridges or storage boxes are bought, while "food" is used for the "normal" consumption. at the end of the year we want to know how much of our membership fees (all in the "membership" class) are remaining. to do that we need to know to which class the category belongs.

Reply:
We also require articles to have a category, and by default there is just an "Other" category. I think we could take the same approach here. And then indeed the feature flag would be if there is more than one supplier category.

Thanks for explaining a bit :+1: